api/schemas/report.json

```diff
@@ -114,6 +114,11 @@
             ]
         },
 
+        "master": {
+            "description": "The name of the master that was used to compile the catalog. If failover occurred, this will hold the first master successfully contacted. If this run had no master (e.g. a `puppet apply` run), this field will be blank",
```

Comment:
Agreed, given the name "master", I'd expect it to always have the name of the master regardless of whether failover happened. (The exception being when masterless (apply))